packages/aws-rfdk/lib/deadline/lib/spot-event-plugin-fleet.ts

```diff
@@ -0,0 +1,575 @@
+ *   bootstrap bucket, or enable AWS CloudTrail on your account to assist in post-incident analysis of compromised production
+ *   environments.
+ * - The data that is stored on your Worker's local EBS volume can include temporary working files from the applications
+ *   that are rendering your jobs and tasks. That data can be sensitive or privileged, so we recommend that you encrypt
+ *   the data volumes of these instances using either the provided option or by using an encrypted AMI as your source.
+ * - The software on the AMI that is being used by this construct may pose a security risk. We recommend that you adopt a
+ *   patching strategy to keep this software current with the latest security patches. Please see
+ *   https://docs.aws.amazon.com/rfdk/latest/guide/patching-software.html for more information.
+ */
+export class SpotEventPluginFleet extends Construct implements ISpotEventPluginFleet {
+  /**
+   * Default prefix for a LogGroup if one isn't provided in the props.
+   */
+  private static readonly DEFAULT_LOG_GROUP_PREFIX: string = '/renderfarm/';
+
+  /**
+   * This is the current maximum for number of workers that can be started on a single host. Currently the
+   * only thing using this limit is the configuration of the listener ports. More than 8 workers can be started,
+   * but only the first 8 will have their ports opened in the workers' security group.
+   */
+  private static readonly MAX_WORKERS_PER_HOST = 8;
+
+  /**
+   * The security groups/rules used to allow network connections.
+   */
+  public readonly connections: Connections;
+
+  /**
+   * The principal to grant permissions to. Granting permissions to this principal will grant
+   * those permissions to the spot instance role.
+   */
+  public readonly grantPrincipal: IPrincipal;
+
+  /**
+   * The port workers listen on to share their logs.
+   */
+  public readonly remoteControlPorts: Port;
+
+  /**
+   * Security Groups assigned to this fleet.
+   */
+  public readonly securityGroups: ISecurityGroup[];
+
+  /**
+   * The user data that instances use when starting up.
+   */
+  public readonly userData: UserData;
+
+  /**
+   * The operating system of the script host.
+   */
+  public readonly osType: OperatingSystemType;
+
+  /**
+   * An IAM role associated with the instance profile assigned to its resources.
+   */
+  public readonly fleetInstanceRole: IRole;
+
+  /**
+   * The IAM instance profile that fleet instance role is associated to.
+   */
+  public readonly instanceProfile: CfnInstanceProfile;
+
+  /**
+   * An IAM role that grants the Spot Fleet the permission to request, launch, terminate, and tag instances on your behalf.
+   */
+  public readonly fleetRole: IRole;
+
+  /**
+   * An id of the Worker AMI.
+   */
+  public readonly imageId: string;
+
+  /**
+   * The tags to apply during creation of instances and of the Spot Fleet Request.
+   */
+  public readonly tags: TagManager;
```

Comment:
This could, and maybe should, be private or protected. Users aren't expected to interact with the `TagManager` instance directly; instead, the expectation is to use the `Tags.of(...).add()` type code.

Reply:
We use `fleet.tags.hasTags` in another construct (see [ConfigureSpotEventPlugin](https://github.com/ryyakobe/aws-rfdk/blob/ryyakobe/sep_configuration/packages/aws-rfdk/lib/deadline/lib/configure-spot-event-plugin.ts#L502)), so I had to make it public.
We had to move the code that converts tags to a different class to bypass JSII limitation for public interfaces having capitalized properties.

In our examples we [demonstrate customers how to use](https://github.com/ryyakobe/aws-rfdk/blob/ryyakobe/sep_configuration/examples/deadline/All-In-AWS-Infrastructure-SEP/ts/lib/sep-stack.ts#L153) `Tags.of()`.

Reply:
Gotcha

Reply:
I could create a separate method in `SpotEventPluginFleet` to encapsulate `hasTags` and `renderTags` if you think it's worth it.